api/fma/v1alpha1/launcherpoolpolicy_types.go

```diff
@@ -0,0 +1,222 @@
+	// AcceleratorType specifies accelerator type (e.g., nvidia.com/gpu)
+	AcceleratorType string `json:"acceleratorType,omitempty"`
+
+	// CountForLauncher is the total number of launcher for each LauncherConfig
+	// to maintain on each matching node per accelerator.
+	// If two different counts are specified for the same (Node, Accelerator, LauncherConfig),
+	// the higher count is used and will be populated into LauncherPoolPolicyStatus.
+	// If no CountForLauncher applies to a given (Node, Accelerator, LauncherConfig), this Node
+	// will be ignored for this LauncherConfig.
+	// +required
+	CountForLauncher []CountForLauncher `json:"countForLauncher"`
+}
+
+// ResourceRequirements defines resource requirements for a node.
+type ResourceRequirements struct {
+	// Allocatable defines the allocatable resources for a node with min/max ranges.
+	// +required
+	Allocatable ResourceRanges `json:"allocatable,omitempty"`
+}
+
+// ResourceRanges defines min/max ranges for various resources of a Node.
+type ResourceRanges struct {
+	// CPU defines the CPU resource range requirement.
+	// +optional
+	CPU ResourceRange `json:"cpu,omitempty"`
+
+	// Memory defines the memory resource range requirement.
+	// +optional
+	Memory ResourceRange `json:"memory,omitempty"`
+
+	// Accelerators defines the GPU resource range requirements keyed by GPU type.
+	// +optional
+	Accelerators map[string]ResourceRange `json:"accelerators,omitempty"`
+}
+
+// ResourceRange defines a range with minimum and maximum quantity values.
+type ResourceRange struct {
+	// Min specifies the minimum quantity required.
+	// +optional
+	Min resource.Quantity `json:"min,omitempty"`
+
+	// Max specifies the maximum quantity allowed.
+	// +optional
+	Max resource.Quantity `json:"max,omitempty"`
+}
+
+// AcceleratorSelector defines accelerator selection criteria
+type AcceleratorSelector struct {
+	// Type specifies accelerator type (e.g., nvidia.com/gpu)
+	Type string `json:"type,omitempty"`
+
+	// Memory specifies accelerator memory size requirement
+	Memory *resource.Quantity `json:"memory,omitempty"`
+
+	// Count specifies required number of accelerators
+	Count *int32 `json:"count,omitempty"`
+}
+
+type CountForLauncher struct {
+	// LauncherConfigName is the name of the LauncherConfig this policy applies to.
+	// +required
+	LauncherConfigName string `json:"launcherConfigName"`
+
+	// LauncherCount is the total number of launcher pods to maintain.
+	// +required
+	LauncherCount int32 `json:"launcherCount"`
+}
+
+type LauncherPoolPolicyStatus struct {
+	// `observedGeneration` is the `metadata.generation` last seen by the controller.
+	// +optional
+	ObservedGeneration int64 `json:"observedGeneration,omitempty"`
+
+	// `errors` reports problems seen in the desired state of this object;
+	// in particular, in the version reported by `observedGeneration`.
+	// +optional
+	Errors []string `json:"errors,omitempty"`
+	// Add status fields if needed (e.g., current idle pod counts)
+}
```

Comment:
It would be good for the Status to also have some information about how many such launcher Pods exist, but I am not sure I know a good way to express that.

One idea would be a potentially large slice giving the count for each (Node, accelerator set, Launcher). But that could get big. Another possibility would be to aggregate over the Nodes, collecting only the min and max. That might require a lot of work every time the controller updates the Status. But maybe it is only iterating over an internal data structure, which should be tolerable.